target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh

```diff
@@ -1,5 +1,129 @@
 REQUIRE_IMAGE_METADATA=1
 
+redmi_ax6000_nand_upgrade_tar()
+{
+	CI_UBIPART=ubi
+	local tar_file="$1"
+	local board_dir="$(tar tf "$tar_file" | grep -m 1 '^sysupgrade-.*/$')"
+	board_dir="${board_dir%/}"
+
+	local kernel_length=$( (tar xf "$tar_file" "$board_dir/kernel" -O | wc -c) 2> /dev/null)
+	[ "$kernel_length" = 0 ] && kernel_length=
+
+	local rootfs_length=$( (tar xf "$tar_file" "$board_dir/root" -O | wc -c) 2> /dev/null)
+	[ "$rootfs_length" = 0 ] && rootfs_length=
+
+	local rootfs_type
+	[ "$rootfs_length" ] && rootfs_type="$(identify_tar "$tar_file" "$board_dir/root")"
+
+	[ -n "$rootfs_length" -o -n "$kernel_length" ] || return 1
+
+	local mtdnum="$( find_mtd_index "$CI_UBIPART" )"
+	if [ ! "$mtdnum" ]; then
+		echo "cannot find ubi mtd partition ubi"
+		return 1
+	fi
+	local ubidev="$( nand_find_ubi "$CI_UBIPART" )"
+	#cleanup old data volume if exist
+	if [ "$ubidev" ] && [ "$( nand_find_volume $ubidev data )" ]; then
+		ubidetach -m "$mtdnum"
+		ubiformat /dev/mtd$mtdnum -y
+		ubiattach -m "$mtdnum"
+		ubidev="$( nand_find_ubi "$CI_UBIPART" )"
+	fi
+	if [ ! "$ubidev" ]; then
+		ubiattach -m "$mtdnum"
+		ubidev="$( nand_find_ubi "$CI_UBIPART" )"
+		if [ ! "$ubidev" ]; then
+			ubiformat /dev/mtd$mtdnum -y
+			ubiattach -m "$mtdnum"
+			ubidev="$( nand_find_ubi "$CI_UBIPART" )"
+
+			if [ ! "$ubidev" ]; then
+				echo "cannot attach ubi mtd partition ubi"
+				return 1
+			fi
+		fi
+	fi
+
+	local kern_mtdnum="$( find_mtd_index "ubi_kernel" )"
```

Comment:
This modification is too risky. Our upgrade process is still different from `nand_upgrade_tar `. Here we need to delete a ubi volume named data, which is somewhat exist at the first time sysupgrade.

Reply:
Duplicating the complete function just for this modification is not acceptable, it may work fine for now, but in the long term we cannot maintain anything like that. I don't see any risk in adding the handling for separate MTD partition holding UBI only used for kernel to the existing nand.sh script. Can you explain which risk you mean?